risc0/zkvm/src/binfmt/image.rs

```diff
@@ -119,23 +120,24 @@ impl MemoryImage {
     /// The result is a MemoryImage with the ELF of `program` loaded (but
     /// execution not yet begun), and with the page table Merkle tree
     /// constructed.
-    pub fn new(program: &Program, page_size: u32) -> Self {
+    pub fn new(program: &Program, page_size: u32) -> Result<Self> {
         let mut buf = vec![0_u8; MEM_SIZE];
 
         // Load the ELF into the memory image.
         for (addr, data) in program.image.iter() {
             let addr = *addr as usize;
             let bytes = data.to_le_bytes();
-            for i in 0..WORD_SIZE {
-                buf[addr + i] = bytes[i];
+            if (WORD_SIZE + addr) >= buf.len() {
+                bail!("Invalid Elf Program, address outside MEM_SIZE");
             }
+            buf[addr..(WORD_SIZE + addr)].copy_from_slice(&bytes[..WORD_SIZE]);
```

Comment:
```suggestion
            buf.get(addr..(WORD_SIZE + addr))?.copy_from_slice(&bytes[..WORD_SIZE]);
```

Reply:
good call, we could drop the if + bail! then right?

Reply:
err I guess that would remove the friendly Error string

Reply:
Combined the two here: https://github.com/risc0/risc0/pull/533/commits/d610dc2fcf909398d3db07bfa3e98d5714615d80